libs/database/src/lib/entities/site-report.entity.ts

```diff
@@ -31,7 +31,8 @@ type ApprovedIdsSubqueryOptions = {
   incomplete: { where: { status: { [Op.notIn]: COMPLETE_REPORT_STATUSES } } },
   sites: (ids: number[] | Literal) => ({ where: { siteId: { [Op.in]: ids } } }),
   approved: { where: { status: { [Op.in]: SiteReport.APPROVED_STATUSES } } },
-  dueBefore: (date: Date | string) => ({ where: { dueAt: { [Op.lt]: date } } })
+  dueBefore: (date: Date | string) => ({ where: { dueAt: { [Op.lt]: date } } }),
+  reportsTask: (taskId: number) => ({ where: { taskId: taskId } })
```

Comment:
I think this would be more consistent if simply named `task`. For instance, that would be in line with the `sites` scope above. This scope is attached to a report, so including the word `report` in the scope is redundant.